src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule DraftEditorCompositionHandlerAndroid
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+import type DraftEditor from 'DraftEditor.react';
+
+const DraftModifier = require('DraftModifier');
+const EditorState = require('EditorState');
+const ReactDOM = require('ReactDOM');
+
+const getDraftEditorSelection = require('getDraftEditorSelection');
+
+let compositionRange = undefined;
+let compositionText = undefined;
+let hasInsertedCompositionText = false;
+let hasMutation = false;
+
+const resetCompositionData = () => {
+  createMutationObserverIfUndefined();
+  compositionRange = undefined;
+  compositionText = undefined;
+  hasInsertedCompositionText = false;
+  hasMutation = false;
+};
+
+const handleMutations = records => {
+  hasMutation = !!records.length;
+  mutationObserver.disconnect();
+};
+
+let mutationObserver = undefined;
+
+// Need to create mutation observer on runtime, not compile time
+const createMutationObserverIfUndefined = () => {
+  if (!mutationObserver) {
+    mutationObserver = new window.MutationObserver(handleMutations);
+  }
+};
+
+/**
+ * Replace the current selection with the specified text string, with the
+ * inline style and entity key applied to the newly inserted text.
+ */
+function replaceText(
+    editorState: EditorState,
+    text: string,
+    compositionRange: SelectionState,
+    // inlineStyle: DraftInlineStyle,
```

Comment:
Don't forget this! I don't think we ever want to pass the entityKey through though.

Reply:
I think we can merge with this, and pick it up later.